nautobot/extras/filters/__init__.py

```diff
@@ -216,12 +218,12 @@ class ConfigContextFilterSet(BaseFilterSet):
     )
     role_id = django_filters.ModelMultipleChoiceFilter(
         field_name="roles",
-        queryset=DeviceRole.objects.all(),
+        queryset=Role.objects.all(),
         label="Role",
     )
     role = django_filters.ModelMultipleChoiceFilter(
         field_name="roles__slug",
-        queryset=DeviceRole.objects.all(),
+        queryset=Role.objects.all(),
         to_field_name="slug",
         label="Role (slug)",
     )
```

Comment:
Possibly collapse these to a single `NaturalKeyOrPKMultipleChoiceFilter` while you're touching this? Or is that out of scope?